app/services/service_provider_updater.rb

```diff
@@ -10,14 +16,18 @@ def run
   def update_local_caches(service_provider)
     issuer = service_provider['issuer']
     if service_provider['active'] == true
-      SERVICE_PROVIDERS[issuer] = service_provider
-      VALID_SERVICE_PROVIDERS << issuer
+      ServiceProvider.find_or_create_by!(issuer: issuer) do |sp|
+        sp.attributes = cleaned_service_provider(service_provider)
+      end
     else
-      SERVICE_PROVIDERS.delete(issuer)
-      VALID_SERVICE_PROVIDERS.delete(issuer)
+      ServiceProvider.destroy_all(issuer: issuer)
     end
   end
 
+  def cleaned_service_provider(service_provider)
+    service_provider.reject { |key, _value| PROTECTED_ATTRIBUTES.include?(key) }
```

Comment:
Rails monkeypatches on a `Hash#except` method:

```ruby
irb(main):003:0> { a: 1, b: 2, c: 3 }.except(:a, :b)
=> {:c=>3}
```

so this could probably just be:

```ruby
service_provider.except(* PROTECTED_ATTRIBUTES)
```